src/main/java/svc/data/citations/filters/CitationFilter.java

```diff
@@ -21,14 +22,24 @@
 		ListIterator<Citation> iter = citations.listIterator();
 		while(iter.hasNext()){
```

Comment:
This might be cleaner with the list/stream operators and some static function that encapsulate the checks you have listed below.

citations.stream().filter(hasWarrantOrCloseCourtDate).collect().toList() or whatever the exact syntax is.